### PR TITLE
More clang-cl.exe warning fixes

### PR DESCRIPTION
**Pull request by shartte**
Remove more unsupported compiler flags when clang-cl.exe is being used.
Introduced new var "CLANG_CL" to indicate that clang-cl.exe is being used and guard compiler switches appropriately.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -145,10 +152,13 @@ if (APPLE)
     # prevents ar from invoking ranlib, let CMake do it
     set(CMAKE_C_ARCHIVE_CREATE   "<CMAKE_AR> qc -S <TARGET> <LINK_FLAGS> <OBJECTS>")
     set(CMAKE_CXX_ARCHIVE_CREATE "<CMAKE_AR> qc -S <TARGET> <LINK_FLAGS> <OBJECTS>")
+elseif(CLANG_CL)
```
Comment (romainguy) [inline CMakeLists.txt]: Add a space between `elseif` and `(`

File: CMakeLists.txt [hunk 2]
```diff
@@ -108,10 +108,17 @@ if ("${CMAKE_C_COMPILER_ID}" STREQUAL "GNU" OR "${CMAKE_CXX_COMPILER_ID}" STREQU
     return()
 endif()
 
+# Detect use of the clang-cl.exe frontend, which does not support all of clangs normal options
+if ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" AND "${CMAKE_CXX_SIMULATE_ID}" STREQUAL "MSVC")
+    set(CLANG_CL true)
```
Comment (bejado) [inline CMakeLists.txt]: Let's reuse the `CLANG_CL` var [here](https://github.com/google/filament/blob/master/filament/CMakeLists.txt#L369)

**Comment by shartte** [comment]
Pushed the requested changes (and fixed the formatting of one more elseif - boy scout rule ;-))

**Comment by romainguy** [comment]
Thanks!

**Pull request merged**